shiny/test/fixture/_fixture.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import logging
+import sys
+from pathlib import PurePath
+from typing import Literal, Union
+
+import pytest
+
+from .._conftest import run_shiny_app
+
+__all__ = (
+    "create_app_fixture",
+    "local_app_fixture_gen",
+    "ScopeName",
+)
+
+
+# Attempt up to 3 times to start the app, with a random port each time
+def local_app_fixture_gen(app: PurePath | str):
```

Comment:
Barret TODO:
* Let's rename this method to `run_shiny_app`. 
* The original `run_shiny_app` should be renamed to an internal method. 
* Expose the original `run_shiny_app` parameters and add `start_attempts = 3` to expose how many start attempts could be made.